compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -987,17 +987,17 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
      * `def hashCode(): Int` and a `static def hashCode(Int): Int`. The JVM
      * back-end considers them as colliding because they have the same name,
      * but we must not.
+     *
+     * Further, unlike scalac, we emit forwarders for *any* static object, not
+     * just top-level ones. This is important so we can implement static methods
+     * of nested static classes of JDK APIs (see #3950).
      */
 
     /** Is the given Scala class, interface or module class a candidate for
      *  static forwarders?
      */
-    def isCandidateForForwarders(sym: Symbol): Boolean = {
-      // it must be a top level class (name contains no $s)
-      !settings.noForwarders && exitingPickler {
-        !sym.name.containsChar('$') && !sym.isNestedClass
-      }
-    }
+    def isCandidateForForwarders(sym: Symbol): Boolean =
+      !settings.noForwarders && sym.isStatic
```

Comment:
We've been using `isStaticModule(sym)` instead of `sym.isStatic` in GenJSCode. I do not remember why `isStaticModule` was needed, but it seems to me that the same condition should be applied in this case.

Reply:
This is tracable back to d217c7d7a084ddc706ad564d799233c493410094.

IIUC we cannot use `isStaticModule`, because this also needs to work for static classes.

Reply:
Oh wow that's old 😄 You're right about the classes, though. Your latest version looks good.